library/common/extensions/filters/http/local_error/filter.cc

```diff
@@ -68,14 +59,16 @@ Http::FilterHeadersStatus LocalErrorFilter::encodeHeaders(Http::ResponseHeaderMa
   }
 
   const auto grpc_status = Grpc::Common::getGrpcStatus(headers);
-  if (grpc_status && grpc_status.value() != Grpc::Status::WellKnownGrpcStatus::Ok) {
+  // gRPC status in headers implies this is an error.
+  if (grpc_status) {
     ASSERT(end_stream,
```

Comment:
we don't need to keep a reference to the headers in the gRPC case _because_ this is the last frame right? Below in the http case we set headers_ to &headers because the filter might stop iteration in order to receive data, right?

Reply:
Right.